src/Package/Impl/ProjectSystem/PropertyPages/Settings/SettingsPageViewModel.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.IO;
+using Microsoft.Common.Core.Shell;
+using Microsoft.R.Components.Application.Configuration;
+using Microsoft.R.Host.Client.Extensions;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.PropertyPages.Settings {
+    internal sealed class SettingsPageViewModel {
+        private readonly Dictionary<string, string> _filesMap = new Dictionary<string, string>();
+        private readonly IConfigurationSettingCollection _settings;
+        private readonly IFileSystem _fileSystem;
+        private readonly ICoreShell _coreShell;
+        private string _currentFile;
+        private string _projectPath;
+
+        public SettingsPageViewModel(IConfigurationSettingCollection settings, ICoreShell coreShell, IFileSystem fileSystem) {
+            _settings = settings;
+            _coreShell = coreShell;
+            _fileSystem = fileSystem;
+        }
+
+        public IEnumerable<string> Files => _filesMap.Keys;
+        public SettingsTypeDescriptor TypeDescriptor => new SettingsTypeDescriptor(_settings);
+
+        public string CurrentFile {
+            get {
+                return _currentFile;
+            }
+            set {
+                if (!value.EqualsIgnoreCase(_currentFile)) {
+                    _currentFile = value;
```

Comment:
We should probably change _currentFile in `finally` block, or restore it in `catch` block